src/Search/Workflow/Blocks.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace eLife\Search\Workflow;
+
+use stdClass;
+
+trait Blocks
+{
+    final private function flattenBlocks(array $blocks) : string
+    {
+        return implode(' ', array_filter(array_map([$this, 'flattenBlock'], $blocks)));
+    }
+
+    final private function flattenBlock(stdClass $block) : string
+    {
+        return implode(' ', array_filter([
+            $block->id ?? null,
+            $block->label ?? null,
+            $block->title ?? null,
+            $block->alt ?? null,
+            $block->text ?? null,
+            $block->caption->text ?? null,
+            $block->question ?? null,
+            $this->flattenBlocks($block->answer ?? []),
+            $this->flattenBlocks($block->content ?? []),
```

Comment:
This isn't every single property that can exist on a block, but I think it covers the important ones.